consensus/tool/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A cli tool for interrogating a mobilecoin consensus node.
+
+#![deny(missing_docs)]
+
+use clap::{Parser, Subcommand};
+use grpcio::{ChannelBuilder, EnvBuilder};
+use mc_common::logger::{create_app_logger, o};
+use mc_connection::BlockInfo;
+use mc_consensus_api::{consensus_common_grpc::BlockchainApiClient, empty::Empty};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConsensusClientUri;
+use serde_json::{to_string_pretty};
+use std::{sync::Arc, time::Duration};
+
+/// A cli tool for interrogating one or more consensus nodes.
+///
+/// Example usage:
+///
+/// $ mc-consensus-tool status mc://localhost:3200 | jq .index
+///
+/// $ mc-consensus-tool status mc://localhost:3200 | jq .network_block_version
+///
+/// $ mc-consensus-tool wait-for-quiet mc://localhost:3200 mc://localhost:3201 mc://localhost:3202
+///
+/// $ MC_PEERS=mc://localhost:3200,mc://localhost:3201 mc-consensus-tool wait-for-block --index=5005
+#[derive(Clone, Debug, Parser)]
+#[clap(name = "mc-consensus-tool")]
+pub struct Config {
+    /// Command to use
+    #[clap(subcommand)]
+    pub tool_command: ToolCommand,
+
+    /// Consensus_uri's to connect to
+    #[clap(global = true, env = "MC_PEERS")]
+    pub consensus_uris: Vec<ConsensusClientUri>,
+}
+
+/// Commands that the tool recognizes
+#[derive(Clone, Debug, Subcommand)]
+pub enum ToolCommand {
+    /// Status: Prints the network status, including block count and block
+    /// version, in json format on STDOUT.
+    Status,
+    /// Wait-for-block: Blocks until all specified nodes have externalized the
+    /// given block index.
+    WaitForBlock {
+        /// Block index which must appear
+        index: u64,
```

Comment:
Don't we need a `clap` annotation here?
```suggestion
        #[clap(long)]
        index: u64,
```